web/crux-ui/src/server/registry-api/google-api-client.ts

```diff
@@ -0,0 +1,87 @@
+import { RegistryImageTags } from '@app/models'
+import { internalError, unauthorizedError } from '@server/error-middleware'
+import { JWT } from 'google-auth-library'
+import { GetAccessTokenResponse } from 'google-auth-library/build/src/auth/oauth2client'
+import { RegistryApiClient } from './registry-api-client'
+
+export type GoogleClientOptions = {
+  username?: string
+  password?: string
+}
+
+export class GoogleRegistryClient implements RegistryApiClient {
+  private headers: HeadersInit = {}
+  private client: JWT
+
+  constructor(private url: string, options?: GoogleClientOptions) {
+    if (options?.username) {
+      if (!options.password) {
```

Comment:
You can wrap this in oneliner if.

Reply:
...Otherhand why are you using optional property accesses in the first and simple check (reverse) in the other if.

Reply:
In my opinion, that's the shortest way to do this check because of the 19. row

Reply:
if (options?.username && !options.password) throw... 

but right now I am thinking about what If both options is undefined?

Reply:
"if (options?.username && !options.password) throw..."  

That's not enough, only need to create the google-auth-client (line 22), if the username and password exist